### PR TITLE
Gh 163 azure aks

### PR DESCRIPTION
**Pull request by nw1984**
PAL-66 merge to develop

### CODE REVIEW

File: service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java [hunk 1]
```diff
@@ -244,8 +243,8 @@ public String toString() {
                     cache(request, futureUser.join(), requestId, multiPolicy, futureResources.join().size(), originalRequestId);
                 })
                 .thenApply(multiPolicy -> {
+                    LOGGER.error("DEBUG: originalRequestId: " + originalRequestId);
```
Comment (ac74475) [inline service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java]: Is this a debugging line?

File: service/data-service/src/main/java/uk/gov/gchq/palisade/data/service/reader/SerialisedDataReader.java [hunk 2]
```diff
@@ -129,6 +130,26 @@ public DataReaderResponse read(final DataReaderRequest request) {
...
+        public void close() throws IOException {
+            try {
+                super.close();
+            } finally {
+                closeAction.run();
+            }
+        }
+    }
+
+
```
Comment (ac74475) [inline service/data-service/src/main/java/uk/gov/gchq/palisade/data/service/reader/SerialisedDataReader.java]: remove extra line

File: example/deployment/local-docker/example-docker-services/resource-service/docker/setenv.sh [hunk 3]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_CONFIG_PATH="/usr/local/tomcat/webapps/resource/WEB-INF/configRest.json"
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/resource-service/docker/setenv.sh]: permissions changed back to remove executable permission, however, the change to add PALISADE_REST_CONFIG_PATH were necessary in that tomcat8 running on azure was unable to recognise the exported variable PALISADE_REST_CONFIG_PATH.

File: example/deployment/local-docker/example-docker-services/user-service/docker/setenv.sh [hunk 4]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_CONFIG_PATH="/usr/local/tomcat/webapps/user/WEB-INF/configRest.json"
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/user-service/docker/setenv.sh]: permissions changed back to remove executable permission, however, the change to add PALISADE_REST_CONFIG_PATH were necessary in that tomcat8 running on azure was unable to recognise the exported variable PALISADE_REST_CONFIG_PATH

File: example/deployment/local-docker/example-docker-services/config-service/docker/setenv.sh [hunk 5]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_BOOTSTRAP_PATH="/usr/local/tomcat/webapps/config/WEB-INF/bootstrapConfig.json"
```
[the remark this answers is not in the document]
Comment (ac74475) [reply example/deployment/local-docker/example-docker-services/config-service/docker/setenv.sh]: Due to the Azure example not picking up the system properties, the containerised deployments will now all put the bootstrap path as a java property instead

File: example/deployment/Azure-AKS/devops-pipelines/azure-pipelines.infra.yaml [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        variables:
+          - group: Palisade-azure
+        steps:
+          - task: charleszipp.azure-pipelines-tasks-terraform.azure-pipelines-tasks-terraform-cli.TerraformCLI@0
+            displayName: 'terraform init'
+            inputs:
+              command: init
+              workingDirectory: '$(System.DefaultWorkingDirectory)/$(azure-dir)/terraform'
+              backendType: azurerm
+              backendServiceArm: 'palisade_rm'
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/Azure-AKS/devops-pipelines/azure-pipelines.infra.yaml]: Creating an azure-pipelines.defaults.yaml template as defined here: https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=example#variable-templates
The variable is as follows:
```
variables:
- name: devOpsServiceConnection
  value: palisade_rm
```

File: example/deployment/Azure-AKS/devops-pipelines/azure-pipelines.k8s.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+stages:
+- stage: Deploy_Palisade
+  jobs:
+    - job:
+      variables: 
+      - group: Palisade-azure
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/Azure-AKS/devops-pipelines/azure-pipelines.k8s.yaml]: Creating an azure-pipelines.defaults.yaml template as defined here: https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=example#variable-templates
The variable is as follows:
```
- name: groupName
  value: Palisade-azure
```

File: example/deployment/local-docker/example-docker-services/config-service/docker/setenv.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/config-service/docker/setenv.sh]: In the AKS environment the tomcat container does not recognise the exported variable, therefore cannot make this change.

File: example/deployment/local-docker/example-docker-services/data-service/docker/setenv.sh [hunk 8]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_CONFIG_PATH="/usr/local/tomcat/webapps/data/WEB-INF/configRest.json"
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/data-service/docker/setenv.sh]: In the AKS environment the tomcat container does not recognise the exported variable, therefore cannot make this change.

File: example/deployment/local-docker/example-docker-services/palisade-service/docker/setenv.sh [hunk 9]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_CONFIG_PATH="/usr/local/tomcat/webapps/palisade/WEB-INF/configRest.json"
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/palisade-service/docker/setenv.sh]: In the AKS environment the tomcat container does not recognise the exported variable, therefore cannot make this change.

File: example/deployment/local-docker/example-docker-services/policy-service/docker/setenv.sh [hunk 10]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 
-export PALISADE_REST_CONFIG_PATH="/usr/local/tomcat/webapps/policy/WEB-INF/configRest.json"
```
[the remark this answers is not in the document]
Comment (nw1984) [reply example/deployment/local-docker/example-docker-services/policy-service/docker/setenv.sh]: In the AKS environment the tomcat container does not recognise the exported variable, therefore cannot make this change.

File: service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java [hunk 11]
```diff
@@ -65,6 +65,9 @@
 import static java.util.Objects.nonNull;
 import static java.util.Objects.requireNonNull;
 
+//import uk.gov.gchq.palisade.service.*;
```
Comment (ac74475) [inline service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java]:
```suggestion
```

File: service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java [hunk 12]
```diff
@@ -65,6 +65,9 @@
...
+//import uk.gov.gchq.palisade.service.request.*;
```
Comment (ac74475) [inline service-impl/palisade-service-impl/simple-palisade-service/src/main/java/uk/gov/gchq/palisade/service/impl/SimplePalisadeService.java]:
```suggestion
```

**Pull request merged**